.travis.yml

```diff
@@ -52,41 +56,53 @@ before_install: |
 
   docker build -t wptd-base -f Dockerfile.base .
   docker build -t "${DOCKER_IMAGE}" -f "${DOCKER_FILE}" .
-  docker run -t -d --entrypoint /bin/bash -v /etc/group:/etc/group:ro -v /etc/passwd:/etc/passwd:ro -v "$(pwd)":/home/jenkins/wpt.fyi -u $(id -u $USER):$(id -g $USER) --name "${DOCKER_INSTANCE}" "${DOCKER_IMAGE}"
-
+  docker run -t -d \
+      --entrypoint /bin/bash \
+      -v /etc/group:/etc/group:ro \
+      -v /etc/passwd:/etc/passwd:ro \
+      -v "$(pwd)":/home/jenkins/wpt.fyi \
+      -u $(id -u $USER):$(id -g $USER) \
+      --name "${DOCKER_INSTANCE}" \
+      "${DOCKER_IMAGE}"
 
-install: |
-  if [ "${WCT}" == "1" ]; then
-    wget -qO- https://raw.githubusercontent.com/creationix/nvm/v0.33.8/install.sh | bash
-    source ~/.nvm/nvm.sh
-    nvm install 6
-    node --version
-    cd webapp
-    npm install -g bower
-    bower install
-    npm install -g web-component-tester
-    npm install
-    cd ../
-    export DISPLAY=:99.0
-    sh -e /etc/init.d/xvfb start
-  fi
 
-  if [ "${MAKE_DEPS_TARGET}" != "" ]; then
-    docker exec -t -u 0:0 "${DOCKER_INSTANCE}" make "${MAKE_DEPS_TARGET}";
-  fi
-
-  docker exec -u 0:0 "${DOCKER_INSTANCE}" chown -R $(id -u $USER):$(id -g $USER) /home/jenkins
+install:
+  - | # Set up Web Components Tests
+    if [ "${WCT}" == "1" ]; then
+      wget -qO- https://raw.githubusercontent.com/creationix/nvm/v0.33.8/install.sh | bash
+      source ~/.nvm/nvm.sh
+      nvm install 6
+      node --version
+      cd webapp
+      npm install -g bower
+      bower install
+      npm install -g web-component-tester
+      npm install
+      cd ../
+      export DISPLAY=:99.0
+      sh -e /etc/init.d/xvfb start
+    fi
+  - | # Set the user as home dir owner in the docker instance
+    docker exec -u 0:0 "${DOCKER_INSTANCE}" chown -R $(id -u $USER):$(id -g $USER) /home/jenkins
+  - | # Install any dependencies
+    if [ "${MAKE_DEPS_TARGET}" != "" ]; then
+      docker exec -t -u 0:0 "${DOCKER_INSTANCE}" make "${MAKE_DEPS_TARGET}" || exit $?
+    fi
 
 script:
-  - |
+  - | # Deploy PR to staging environment
+    if [ "${MAKE_DEPLOY_TARGET}" != "" ]; then
+      echo "Copying output to ${TEMP_FILE:=$(mktemp)}"
+      docker exec -t -u $(id -u $USER):$(id -g $USER) "${DOCKER_INSTANCE}" make "${MAKE_DEPLOY_TARGET}" BRANCH_NAME=${TRAVIS_PULL_REQUEST_BRANCH} 2>&1 | tee ${TEMP_FILE}
+      if [ "${EXIT_CODE:=${PIPESTATUS[0]}}" != "0" ]; then exit ${EXIT_CODE}; fi
+      DEPLOYED_URL="$(grep -Po 'Deployed to \K.*$' ${TEMP_FILE})"
```

Comment:
What about

```bash
grep -Po 'Deployed to \K\S$' ${TEMP_FILE}
```

`\S` for non whitespace, so that we don't need to `tr` in `deploy-comment.sh`. This would be more intuitive and future-proof IMHO. For example, if later we want to use `${DEPLOYED_URL}` in some other command/script, we don't need to trim it again.

Reply:
grep prints a trailing newline. Moved the `tr` into .travis.yml though.